tests/unit/sagemaker/serve/builder/test_js_builder.py

```diff
@@ -0,0 +1,470 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+from unittest.mock import MagicMock, patch
+
+import unittest
+
+from sagemaker.serve.builder.model_builder import ModelBuilder
+from sagemaker.serve.mode.function_pointers import Mode
+from sagemaker.serve.utils.exceptions import (
+    LocalDeepPingException,
+    LocalModelLoadException,
+    LocalModelOutOfMemoryException,
+    LocalModelInvocationException,
+)
+
+mock_model_id = "huggingface-llm-amazon-falconlite"
+mock_t5_model_id = "google/flan-t5-xxl"
+mock_prompt = "Hello, I'm a language model,"
+mock_response = "Hello, I'm a language model, and I'm here to help you with your English."
+mock_sample_input = {"inputs": mock_prompt, "parameters": {}}
+mock_sample_output = [{"generated_text": mock_response}]
+
+mock_set_serving_properties = (4, "fp16", 1, 256, 256)
+
+mock_tgi_most_performant_model_serving_properties = {
+    "SAGEMAKER_PROGRAM": "inference.py",
+    "SAGEMAKER_MODEL_SERVER_WORKERS": "1",
+    "SM_NUM_GPUS": "4",
+}
+mock_tgi_model_serving_properties = {
+    "SAGEMAKER_PROGRAM": "inference.py",
+    "SAGEMAKER_MODEL_SERVER_WORKERS": "1",
+    "SM_NUM_GPUS": "2",
+}
+
+mock_djl_most_performant_model_serving_properties = {
+    "SAGEMAKER_PROGRAM": "inference.py",
+    "SAGEMAKER_MODEL_SERVER_WORKERS": "1",
+    "OPTION_TENSOR_PARALLEL_DEGREE": "4",
+}
+mock_djl_model_serving_properties = {
+    "SAGEMAKER_PROGRAM": "inference.py",
+    "SAGEMAKER_MODEL_SERVER_WORKERS": "4",
+    "OPTION_TENSOR_PARALLEL_DEGREE": "4",
+}
+
+mock_schema_builder = MagicMock()
+mock_schema_builder.sample_input = mock_sample_input
+mock_schema_builder.sample_output = mock_sample_output
+
+mock_tgi_image_uri = (
+    "123456789712.dkr.ecr.us-west-2.amazonaws.com/huggingface-pytorch-tgi"
+    "-inference:2.1.1-tgi1.4.0-gpu-py310-cu121-ubuntu20.04"
+)
+mock_djl_image_uri = (
+    "123456789712.dkr.ecr.us-west-2.amazonaws.com/djl-inference:0.24.0-neuronx-sdk2.14.1"
+)
+
+
+class TestJumpStartBuilder(unittest.TestCase):
+    @patch("sagemaker.serve.builder.jumpstart_builder._capture_telemetry", side_effect=None)
```

Comment:
Optional: Using this might help clean up these patch(s) : https://docs.python.org/3.10/whatsnew/3.10.html#parenthesized-context-managers